pulpcore/cli/common/generic.py

```diff
@@ -490,7 +490,11 @@ def _option_callback(
 
         if value.startswith("/"):
             # The HREF of a resource was passed
-            pattern = rf"^{pulp_ctx.api_path}{entity_ctx.HREF_PATTERN}"
+            href_pattern = entity_ctx.HREF_PATTERN
+            if pulp_ctx.domain_enabled:
+                pattern = rf"^{pulp_ctx._api_root}(?P<pulp_domain>\w+)/api/v3/{href_pattern}"
```

Comment:
Why is this needed? We only accept entities of the proper domain anyway. If the domain in hrefs is optional (I don't think it is), we can still do `rf"^{pulp_ctx._api_root}(?{pulp_ctx.domain_name}/)api/v3/{href_pattern}"` right?

Reply:
This is for user/group role assignment. Assigning an object-level role can have the object be in a different domain than the user/group which are always in the default domain.